tests/test_ensemble.py

```diff
@@ -0,0 +1,158 @@
+
+        #custom conjugates set, allowing both mu and N
+        ens = relentless.Ensemble(types=('A','B'), T=100, V=3.0, mu={'A':0.1,'B':0.2}, N={'A':1,'B':2}, conjugates=('P',))
+        self.assertEqual(ens.P, None)
+        self.assertAlmostEqual(ens.V, 3.0)
+        self.assertDictEqual(ens.mu, {'A':0.1,'B':0.2})
+        self.assertDictEqual(ens.N, {'A':1,'B':2})
+        self.assertCountEqual(ens.conjugates, ('P',))
+
+        #custom conjugates set, allowing P, V, mu, and N
+        ens = relentless.Ensemble(types=('A','B'), T=100, P=1.5, V=3.0, mu={'A':0.1,'B':0.2}, N={'A':1,'B':2}, conjugates=('q',))
+        self.assertEqual(ens.P, 1.5)
+        self.assertAlmostEqual(ens.V, 3.0)
+        self.assertDictEqual(ens.mu, {'A':0.1,'B':0.2})
+        self.assertDictEqual(ens.N, {'A':1,'B':2})
+        self.assertCountEqual(ens.conjugates, ('q',))
+
+        #test creation with mu and N dicts containing types not in ensemble
+        ens = relentless.Ensemble(types=('A','B'), T=100, V=3.0, mu={'A':0.1,'C':0.2}, N={'B':2,'D':3})
+        self.assertCountEqual(ens.types, ('A','B'))
+        self.assertCountEqual(ens.rdf.pairs, (('A','B'),('A','A'),('B','B')))
+        self.assertAlmostEqual(ens.kB, 1.0)
+        self.assertAlmostEqual(ens.T, 100)
+        self.assertEqual(ens.P, None)
+        self.assertAlmostEqual(ens.V, 3.0)
+        self.assertDictEqual(ens.mu, {'A':0.1,'B':None})
+        self.assertDictEqual(ens.N, {'A':None,'B':2})
+        self.assertCountEqual(ens.conjugates, ('P','mu_B','N_A'))
+        self.assertAlmostEqual(ens.beta, 0.01)
+
+        #test creation with single type
+        ens = relentless.Ensemble(types=('A',), T=100, V=3.0, mu={'A':0.1})
+        self.assertCountEqual(ens.types, ('A',))
+        self.assertCountEqual(ens.rdf.pairs, (('A','A'),))
+        self.assertAlmostEqual(ens.kB, 1.0)
+        self.assertAlmostEqual(ens.T, 100)
+        self.assertEqual(ens.P, None)
+        self.assertAlmostEqual(ens.V, 3.0)
+        self.assertDictEqual(ens.mu, {'A':0.1})
+        self.assertDictEqual(ens.N, {'A':None})
+        self.assertCountEqual(ens.conjugates, ('P','N_A'))
+        self.assertAlmostEqual(ens.beta, 0.01)
+
+        #test invalid setting of neither P nor V
+        with self.assertRaises(ValueError):
+            ens = relentless.Ensemble(types=('A','B'), T=10, mu={'A':0.1,'B':0.2})
+
+        #test invalid setting of neither mu nor N
+        with self.assertRaises(ValueError):
+            ens = relentless.Ensemble(types=('A','B'), T=10, P=2.0)
+
+        #conjugates not set, test invalid setting of both P and V
+        with self.assertRaises(ValueError):
+            ens = relentless.Ensemble(types=('A','B'), T=10, P=2.0, V=3.0)
+
+        #conjugates not set, test invalid setting of both mu and N
+        with self.assertRaises(ValueError):
+            ens = relentless.Ensemble(types=('A','B'), T=10, mu={'A':0.1,'B':0.2}, N={'A':1,'B':2})
+
```

Comment:
Pending what we decide on the behavior of `conjugates`, do we include a test where both `P` and `V` (or `mu` and `N`) are included in `conjugates`?